cmd/grpcurl/grpcurl.go

```diff
@@ -363,6 +363,9 @@ func main() {
 	if *format != "json" && *format != "text" {
 		fail(nil, "The -format option must be 'json' or 'text'.")
 	}
+	if isUnixSocket != nil && isUnixSocket() {
+		fail(nil, "The -unix option is deprecated, please use unix://{path} as the address instead.")
+	}
```

Comment:
I don't think it should fail the command. That would be backwards-incompatible. Instead, it should add the "unix://" prefix to the target addres (if it's not already present). I don't actually think we even need to deprecate the flag -- we could just keep it as a synonym of sorts for using a "unix://" prefix. (See other comment for more.)